### PR TITLE
MSONAR-207 The scanner for Maven collects files outside of conventional sonar.sources

### PR DESCRIPTION
**Pull request by dorian-burihabwa-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/FileVisitorImpl.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+/**
+sonar.source = dir/intersted.py
+sonar.exclusions = dir/not-interested.rs
+
+dir/
+  interested.py
+  not-interseted.rs
+*/
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/FileVisitorImpl.java:148]: Remove dangling comment

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java [hunk 2]
```diff
@@ -122,6 +123,18 @@ public class MavenProjectConverter {
...
 
+  public Set<Path> getSkippedBasedDirs() {
+    return skippedBasedDirs;
+  }
+
+  private final Set<Path> skippedBasedDirs = new HashSet<>();
+
+  private boolean sourceDirsIsOverridden = false;
+
+  public boolean isSourceDirsIsOverridden() {
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java:134]: Rename to be less redundant
```suggestion
  public boolean isSourceDirsOverridden() {
```

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java [hunk 3]
```diff
@@ -122,6 +123,18 @@ public class MavenProjectConverter {
 
   private final MavenCompilerResolver mavenCompilerResolver;
 
+  public Set<Path> getSkippedBasedDirs() {
+    return skippedBasedDirs;
+  }
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java:128]: Move the getters below

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java [hunk 4]
```diff
@@ -529,13 +543,14 @@ private List<File> testSources(MavenProject pom) throws MojoExecutionException {
   private List<File> sourcePaths(MavenProject pom, String propertyKey, Collection<String> mavenPaths) throws MojoExecutionException {
     List<File> filesOrDirs;
     boolean userDefined = false;
-    String prop = StringUtils.defaultIfEmpty(userProperties.getProperty(propertyKey), envProperties.getProperty(propertyKey));
-    prop = StringUtils.defaultIfEmpty(prop, pom.getProperties().getProperty(propertyKey));
+    String prop = getProperty(pom, propertyKey);
 
+    // User property > Environment property > Pom
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java:548]: Delete comment
```suggestion
```

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java [hunk 5]
```diff
@@ -551,6 +566,12 @@ private List<File> sourcePaths(MavenProject pom, String propertyKey, Collection<
     }
   }
 
+  public String getProperty(MavenProject pom, String propertyKey) {
+    String prop = StringUtils.defaultIfEmpty(userProperties.getProperty(propertyKey), envProperties.getProperty(propertyKey));
+    prop = StringUtils.defaultIfEmpty(prop, pom.getProperties().getProperty(propertyKey));
+    return prop;
+  }
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenProjectConverter.java:573]: Revert method extraction

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenUtils.java [hunk 6]
```diff
@@ -113,4 +116,38 @@ private static String escapeCommas(String value) {
     return value.contains(",") ? ("\"" + value + "\"") : value;
   }
 
+  public static List<String> splitAsCsv(String joined) {
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/MavenUtils.java:119]: We use StringUtils elsewhere in the code. Can we replace the custom implem with a method call?

File: src/main/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapper.java [hunk 7]
```diff
@@ -62,12 +71,14 @@ public void execute() throws MojoExecutionException {
       }
 
       scanner.execute(collectProperties());
+      // Extra
```
Comment (dorian-burihabwa-sonarsource) [inline src/main/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapper.java:74]: Delete this
```suggestion
```

**Review by leveretka — APPROVED** [review]
LGTM!

**Pull request merged**